internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepository.java

```diff
@@ -128,11 +129,77 @@ private Object doSchemaOperation(String storeName, Function<SchemaData, Object>
     }
   }
 
-  private void mayRegisterAndPopulateMetadataSchema(Store store, SchemaData schemaData) {
+  void maybeRegisterAndPopulateRmdSchema(Store store, SchemaData schemaData) {
     if (store.isActiveActiveReplicationEnabled()) {
       String storeName = store.getName();
-      accessor.subscribeReplicationMetadataSchemaCreationChange(storeName, replicationMetadataSchemaChildListener);
-      accessor.getAllReplicationMetadataSchemas(storeName).forEach(schemaData::addReplicationMetadataSchema);
+      getAccessor().subscribeReplicationMetadataSchemaCreationChange(storeName, replicationMetadataSchemaChildListener);
+      getAccessor().getAllReplicationMetadataSchemas(storeName).forEach(schemaData::addReplicationMetadataSchema);
+    }
+  }
+
+  void maybeRegisterAndPopulateUpdateSchema(Store store, SchemaData schemaData) {
+    if (store.isWriteComputationEnabled()) {
+      String storeName = store.getName();
+      getAccessor().subscribeDerivedSchemaCreationChange(storeName, derivedSchemaChildListener);
+      getAccessor().getAllDerivedSchemas(storeName).forEach(schemaData::addDerivedSchema);
+    }
+  }
+
+  void maybeForceRefreshSchemaDataForSupersetSchemaWithRetry(Store store, SchemaData schemaData, int supersetSchemaId) {
+    final int maxRefreshAttempt = 3;
+    int refreshAttempt = 0;
+    while (true) {
```

Comment:
Shall we use `RetryUtils` instead?

Reply:
Sure.. I was thinking RetryUtils does not provide a pre-check before first retry but I guess I can handle that by wrapping the check logic into a separated method.